sui_storage/src/lock_service.rs

```diff
@@ -0,0 +1,552 @@
+        info!("LockService command processing loop started");
+        // NOTE: we use blocking_recv() as its faster than using regular async recv() with awaits in a loop
+        while let Some(msg) = receiver.blocking_recv() {
+            match msg {
+                LockServiceCommands::Acquire {
+                    refs,
+                    tx_digest,
+                    resp,
+                } => {
+                    let res = self.acquire_locks(&refs, tx_digest);
+                    if let Err(_e) = resp.send(res) {
+                        info!("Could not respond to sender, sender dropped!");
+                    }
+                }
+                LockServiceCommands::Initialize { refs, resp } => {
+                    if let Err(_e) = resp.send(self.initialize_locks(&refs)) {
+                        info!("Could not respond to sender, sender dropped!");
+                    }
+                }
+                LockServiceCommands::RemoveLocks { refs, resp } => {
+                    if let Err(_e) = resp.send(self.delete_locks(&refs)) {
+                        info!("Could not respond to sender, sender dropped!");
+                    }
+                }
+            }
+        }
+        info!("LockService command loop stopped, the sender on other end hung up/dropped");
+    }
+
+    /// Loop to continuously process queries in a single thread
+    fn queries_loop(&self, mut receiver: Receiver<LockServiceQueries>) {
+        info!("LockService queries processing loop started");
+        while let Some(msg) = receiver.blocking_recv() {
+            match msg {
+                LockServiceQueries::GetLock { object, resp } => {
+                    if let Err(_e) = resp.send(self.get_lock(object)) {
+                        info!("Could not respond to sender!");
+                    }
+                }
+                LockServiceQueries::CheckLocksExist { objects, resp } => {
+                    if let Err(_e) = resp.send(self.locks_exist(&objects)) {
+                        info!("Could not respond to sender, sender dropped!");
+                    }
+                }
+            }
+        }
+        info!("LockService queries loop stopped, the sender on other end hung up/dropped");
+    }
+}
+
+const LOCKSERVICE_QUEUE_LEN: usize = 500;
+
+/// Atomic Sui Object locking service.
+/// Primary abstraction is an atomic op to acquire a lock on a given set of objects.
+/// Atomicity relies on single threaded loop and only one instance per authority.
+#[derive(Clone)]
+pub struct LockService {
+    sender: Sender<LockServiceCommands>,
+    query_sender: Sender<LockServiceQueries>,
+}
+
+impl LockService {
+    /// Create a new instance of LockService.  For now, the caller has to guarantee only one per data store -
+    /// namely each SuiDataStore creates its own LockService.
+    pub fn new<P: AsRef<Path>>(path: P, db_options: Option<Options>) -> Result<Self, SuiError> {
```

Comment:
Same thing as the remark about Narwhal conventions above: this one would be called `spawn`.